product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -21,44 +21,48 @@ def _name_search(
         res = super(ProductProduct, self)._name_search(
             name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
         )
-        if not limit or len(res) >= limit:
-            limit = (limit - len(res)) if limit else False
+        res_ids = list(res)
+        res_ids_len = len(res_ids)
+        if not limit or res_ids_len >= limit:
+            limit = (limit - res_ids_len) if limit else False
         if (
             not name
             and limit
             or not self._context.get("partner_id")
-            or len(res) >= limit
+            or res_ids_len >= limit
         ):
-            return res
-        limit -= len(res)
-        customerinfo_ids = self.env["product.customerinfo"]._search(
-            [
-                ("name", "=", self._context.get("partner_id")),
-                "|",
-                ("product_code", operator, name),
-                ("product_name", operator, name),
-            ],
-            limit=limit,
-            access_rights_uid=name_get_uid,
+            return res_ids
+        limit -= res_ids_len
+        customerinfo_ids = list(
+            self.env["product.customerinfo"]._search(
+                [
+                    ("name", "=", self._context.get("partner_id")),
+                    "|",
+                    ("product_code", operator, name),
+                    ("product_name", operator, name),
+                ],
+                limit=limit,
+                access_rights_uid=name_get_uid,
+            )
         )
         if not customerinfo_ids:
-            return res
-        res_templates = self.browse([product_id for product_id, _name in res]).mapped(
-            "product_tmpl_id"
-        )
+            return res_ids
+        res_templates = self.browse(res_ids).mapped("product_tmpl_id")
         product_tmpls = (
             self.env["product.customerinfo"]
             .browse(customerinfo_ids)
             .mapped("product_tmpl_id")
             - res_templates
         )
-        product_ids = self._search(
-            [("product_tmpl_id", "in", product_tmpls.ids)],
-            limit=limit,
-            access_rights_uid=name_get_uid,
+        product_ids = list(
+            self._search(
```

Comment:
Shouldn't `_self` already return list of ids? (Ref. https://github.com/odoo/odoo/blob/94944e1ed0f3cd7d344ab42788e770ad58d8a49c/odoo/models.py#L4482)

Reply:
I found that **_search** method returns a **odoo.osv.query.Query** object. 
here I need to ids so I can use list() or **odoo.osv.query.Query._result** (Ref. https://github.com/odoo/odoo/blob/94944e1ed0f3cd7d344ab42788e770ad58d8a49c/odoo/osv/query.py#L147). let me know what do you prefer.

anyway I changed here https://github.com/OCA/product-attribute/pull/859/files#diff-f00b983e1fac4d29d20ae66593f2df2378eb5445c33b11b2ea43421a0a81e463R36 from previous version as **odoo.osv.query.Query** object can be directly pass to **browse** method